code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -262,6 +261,42 @@
 		var/preferred_icon = input ? input : C.prefs.read_preference(/datum/preference/choiced/ai_core_display)
 		icon_state = resolve_ai_icon(preferred_icon)
 
+/// Apply an AI's hologram preference
+/mob/living/silicon/ai/proc/apply_pref_hologram_display(client/player_client)
+	if(player_client.prefs?.read_preference(/datum/preference/choiced/ai_hologram_display))
+		var/list/hologram_choice = player_client.prefs.read_preference(/datum/preference/choiced/ai_hologram_display)
+		if(hologram_choice == "Random")
+			hologram_choice = pick(GLOB.ai_hologram_icons)
+
+		hologram_appearance = mutable_appearance(GLOB.ai_hologram_icons[hologram_choice], GLOB.ai_hologram_icon_state[hologram_choice])
+
+	hologram_appearance ||= mutable_appearance(GLOB.ai_hologram_icons[AI_HOLOGRAM_DEFAULT], GLOB.ai_hologram_icon_state[AI_HOLOGRAM_DEFAULT])
+
+/// Apply an AI's emote display preference
+/mob/living/silicon/ai/proc/apply_pref_emote_display(client/player_client)
+	if(player_client.prefs?.read_preference(/datum/preference/choiced/ai_emote_display))
+		var/emote_choice = player_client.prefs.read_preference(/datum/preference/choiced/ai_emote_display)
+
+		if(emote_choice == "Random")
+			emote_choice = pick(GLOB.ai_status_display_emotes)
+
+		apply_emote_display(emote_choice)
+
+/// Apply an emote to all AI status displays on the station
+/mob/living/silicon/ai/proc/apply_emote_display(emote)
+	var/turf/ai_turf = get_turf(loc)
+
+	for(var/obj/machinery/status_display/ai/ai_display as anything in GLOB.ai_status_displays)
+		var/turf/display_turf = get_turf(ai_display)
+
+		// - Station AIs can change every display on the station Z.
+		// - Ghost role AIs (or AIs on the mining base?) can only affect their Z
+		if(!is_valid_z_level(ai_turf, display_turf))
+			continue
```

Comment:
I think this should only work if the AI is on the station z level, because if we have any AI displays in the mining base then they will never be set unless a ghost role does it, which i think it's inconsistent since the mining base is "part of the station".

Reply:
This preference is only triggered when you join using the AI job.  Any newly created AIs during the round won't have this triggered.

Reply:
The comment mislead me I guess, I thought this would be triggered when someone would be turned into an AI.
My comment still stands however, this means that the station AI won't change any AI displays in the mining base

Reply:
Applied with 507d78cf45f7ccbb77f50dc0c0768a0378caa0ad